webpush/encrypt_test.go

```diff
@@ -135,19 +139,78 @@ func TestEncrypt(t *testing.T) {
 		t.Errorf("Expected ciphertext to be %v, got %v", result.Ciphertext, expCiphertext)
 	}
 
-	_, expKey, err := generateMockKeys()
+	_, expKey, err := mockKeys()
 	if err != nil {
 		t.Error(err)
 	}
 	if !bytes.Equal(result.ServerPublicKey, expKey) {
 		t.Errorf("Expected server key to be %v, got %v", result.ServerPublicKey, expKey)
 	}
 
-	expSalt, err := generateMockSalt()
+	expSalt, err := mockSalt()
 	if err != nil {
 		t.Error(err)
 	}
 	if !bytes.Equal(result.Salt, expSalt) {
 		t.Errorf("Expected salt to be %v, got %v", result.Salt, expSalt)
 	}
 }
+
+func rfcSalt() ([]byte, error) {
+	return base64.URLEncoding.WithPadding(base64.NoPadding).DecodeString("lngarbyKfMoi9Z75xYXmkg")
+}
+
+func rfcKeys() ([]byte, []byte, error) {
+	priv, _ := base64.URLEncoding.WithPadding(base64.NoPadding).DecodeString("nCScek-QpEjmOOlT-rQ38nZzvdPlqa00Zy0i6m2OJvY")
+
+	// Generate the right public key for the static private key
+	x, y := curve.ScalarMult(curve.Params().Gx, curve.Params().Gy, priv)
+
+	return priv, elliptic.Marshal(curve, x, y), nil
+}
+
+// TestRfcVectors uses the values given in the RFC for HTTP encryption to verify
+// that the code conforms to the RFC
+// See: https://tools.ietf.org/html/draft-ietf-httpbis-encryption-encoding-02#appendix-B
+func TestRfcVectors(t *testing.T) {
+	defer stubFuncs(rfcSalt, rfcKeys)()
+
+	b64 := base64.URLEncoding.WithPadding(base64.NoPadding)
+
+	auth, err := b64.DecodeString(rfcAuth)
+	if err != nil {
+		t.Error(err)
+	}
+	key, err := b64.DecodeString(rfcPublic)
+	if err != nil {
+		t.Error(err)
+	}
+
+	sub := &Subscription{Auth: auth, Key: key}
+
+	result, err := Encrypt(sub, message)
+	if err != nil {
+		t.Error(err)
+	}
+
+	expCiphertext, err := b64.DecodeString(rfcCipher)
+	if err != nil {
+		t.Error(err)
+	}
+	if !bytes.Equal(result.Ciphertext, expCiphertext) {
+		t.Errorf("Expected ciphertext to be %v, got %v", result.Ciphertext, expCiphertext)
```

Comment:
Not sure what you mean. Is this about the wording of the error? Something like "Ciphertext: have: %v, want: %v"?